src/app/client/src/app/modules/public/components/signup/signup.component.spec.ts

```diff
@@ -2,31 +2,128 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { SignupComponent } from './signup.component';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
 import { SuiModule } from 'ng2-semantic-ui';
-import { FormBuilder } from '@angular/forms';
+import { FormBuilder, FormGroup } from '@angular/forms';
 import { ResourceService, SharedModule } from '@sunbird/shared';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { SignupService } from '../../services/signup.service';
+import { Router } from '@angular/router';
+import { mockSignUpResponse } from './signup.component.spec.data';
+import { Observable } from 'rxjs/Observable';
+import { CoreModule, LearnerService } from '@sunbird/core';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 
 describe('SignupComponent', () => {
   let component: SignupComponent;
   let fixture: ComponentFixture<SignupComponent>;
 
+  class RouterStub {
+    navigate = jasmine.createSpy('navigate');
+  }
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ SignupComponent ],
-      imports: [SuiModule, SharedModule, HttpClientTestingModule],
-      providers: [FormBuilder, ResourceService],
+      declarations: [SignupComponent],
+      imports: [FormsModule, ReactiveFormsModule, SuiModule, SharedModule, CoreModule, HttpClientTestingModule],
+      providers: [FormBuilder, ResourceService, SignupService, LearnerService,
+        { provide: Router, useClass: RouterStub }],
       schemas: [NO_ERRORS_SCHEMA]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SignupComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should call redirect method', () => {
+    const router = TestBed.get(Router);
+    component.redirect();
+    expect(router.navigate).toHaveBeenCalledWith(['']);
+  });
+  it('should show validation error message for form', () => {
```

Comment:
all this spec should be tested after changing input and should check error message as well. Discusses this with me